lib/deimos/active_record_consume/batch_consumption.rb

```diff
@@ -142,24 +159,74 @@ def save_records_to_database(record_class, key_cols, records)
       def import_associations(entities)
         _validate_associations(entities)
         _fill_primary_key_on_entities(entities)
+        keys = []
 
         # Select associations from config parameter association_list and
         # fill id to associated_objects foreign_key column
         @klass.reflect_on_all_associations.select { |assoc| @association_list.include?(assoc.name) }.
           each do |assoc|
+
+            records_to_remove = []
             sub_records = entities.map { |entity|
               # Get associated `has_one` or `has_many` records for each entity
               sub_records = Array(entity.send(assoc.name))
               # Set IDS from master to each of the records in `has_one` or `has_many` relation
+
+              new_records = sub_records.map { |record| record.attributes['id'] == nil ? record : nil }.compact
+              old_records = sub_records.map { |record| record.attributes['id'] != nil ? record : nil }.compact
```

Comment:
This assumes that the records coming in would have an ID. In the offer/product use case, there will **never** be an ID because the downstream system doesn't "own" the data, it's only requesting you to update the data that you own. I don't think this will work either. :disappointed: I think there is a solution we can use - will chat about it once PDA-114 is done.

Reply:
Sounds great, we can chat about that after PDA-114